src/main/kotlin/com/openlattice/shuttle/payload/OrcFilesPayload.kt

```diff
@@ -0,0 +1,185 @@
+    }
+}
+
+class OrcIterator(val path: Path): Iterator<Map<String, Any?>> {
+
+    private val orcConfiguration = Configuration()
+    private var currentBatch: VectorizedRowBatch
+    private var rows: RecordReader
+    private var currentBuffer: Array<MutableMap<String, Any?>> = emptyArray()
+    private var currentBufferIndex: Int
+    private var totalColumnCount: Int
+    private var fieldNames: List<String>
+
+    init {
+        orcConfiguration.setBoolean("header", true)
+        orcConfiguration.setBoolean("inferSchema", true)
+    }
+
+    val orcReader = OrcFile.createReader(
+            org.apache.hadoop.fs.Path(path.toUri()),
+            OrcFile.ReaderOptions(orcConfiguration)
+    )
+
+    init {
+        rows = orcReader.rows()
+        currentBatch = orcReader.schema.createRowBatch()
+        fieldNames = orcReader.schema.fieldNames
+        currentBufferIndex = 0
+        rows.nextBatch( currentBatch )
+        totalColumnCount = currentBatch.numCols
+        currentBuffer = mapBatchToRealTypes(currentBatch.size)
+    }
+
+    override fun hasNext(): Boolean {
+        return !currentBatch.endOfFile
+    }
+
+    companion object {
+        fun processList( columnVector: ColumnVector ): List<Any>{
+            when (columnVector.type) {
+                ColumnVector.Type.LONG -> {
+                    return Lists.newArrayList( (columnVector as LongColumnVector).vector )
+                }
+                ColumnVector.Type.DOUBLE -> {
+                    return Lists.newArrayList( (columnVector as DoubleColumnVector).vector )
+                }
+                ColumnVector.Type.BYTES -> {
+                    return Lists.newArrayList( (columnVector as BytesColumnVector).vector )
+                }
+                ColumnVector.Type.DECIMAL -> {
+                    return Lists.newArrayList( (columnVector as DecimalColumnVector).vector )
+                }
+                ColumnVector.Type.DECIMAL_64 -> {
+                    return Lists.newArrayList( (columnVector as Decimal64ColumnVector).vector )
+                }
+                ColumnVector.Type.LIST -> {
+                    return processList( (columnVector as ListColumnVector ))
+                }
+                ColumnVector.Type.TIMESTAMP,
+                ColumnVector.Type.INTERVAL_DAY_TIME,
+                ColumnVector.Type.STRUCT,
+                ColumnVector.Type.MAP,
+                ColumnVector.Type.UNION,
+                ColumnVector.Type.VOID,
+                ColumnVector.Type.NONE -> {
+                }
+            }
+            return ImmutableList.of()
+        }
+
+        fun processColumnVectorType(columnVector: ColumnVector, newBufferSize: Int, columnName: String, buffer: Array<MutableMap<String, Any?>>){
+            println("it a ${columnVector.type}")
+            when (columnVector.type) {
+                ColumnVector.Type.LONG -> {
+                    val longVector = (columnVector as LongColumnVector).vector
```

Comment:
Might be simpler to introduce some kind of `getTypedVector` function that does this switch statement and returns `(columnVector as BlahVector).vector` -- then you'd only need to handle `ColumnVector.Type.LIST` differently here, and you could collapse the logic for all the types that return a new array list above in `processList`